core/src/main/java/org/apache/gravitino/storage/external/stats/PartitionStatisticStorage.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.storage.external.stats;
+
+import com.google.common.collect.Lists;
+import java.io.Closeable;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.stats.PartitionRange;
+import org.apache.gravitino.stats.PartitionStatisticsDrop;
+import org.apache.gravitino.stats.PartitionStatisticsUpdate;
+import org.apache.gravitino.stats.StatisticValue;
+
+/** Interface for managing partition statistics in a storage system. */
+public interface PartitionStatisticStorage extends Closeable {
+
+  /**
+   * Returns the name of the metalake associated with this storage.
+   *
+   * @return the name of the metalake
+   */
+  String metalake();
+
+  /**
+   * Returns the properties of this storage.
+   *
+   * @return a map of properties associated with this storage.
+   */
+  Map<String, String> properties();
```

Comment:
Why do you need this two interface, can you provide a mock implementation, which can also used for test.

Reply:
OK. I have implemented Memory storage and LanceDB storage. I can submit memory storage first.

Actually, I added these two interfaces to constrain the storage field. We should have metalake and properties as the class fields..
Metalake field: In other interfaces, because we don't need to pass metalake as the paramter.
properties: Upper layer can get storage property, such as location, etc.